### PR TITLE
Create slides and samples for synchronization

### PR DESCRIPTION
**Pull request by linuxmonk**
Addresses issue #83 

https://pr86-dot-gotraining-testing.appspot.com/gocourse-2019-05-melbourne.slide#70

### CODE REVIEW

File: content/gocourse-2019-05-melbourne.slide [hunk 1]
```diff
@@ -841,32 +841,16 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
...
-- [[https://goplay.space/#ANO8rpfNynY][Range and close]] ([[https://play.golang.org/p/ANO8rpfNynY][Playground]]`*`)
-- [[https://goplay.space/#R1qgjOsPSGk][Select]] ([[https://play.golang.org/p/R1qgjOsPSGk][Playground]]`*`)
-- [[https://goplay.space/#4TGM29YbvX4][Deadlocks]] ([[https://play.golang.org/p/4TGM29YbvX4][Playground]]`*`)
-
-`*` The [[https://play.golang.org/][official Go Playground]] prints in real-time with delays but lacks syntax colouring.
-
+* Synchronization
+
+- Available in [[https://golang.org/pkg/sync/][`sync`]] package
+- High level Synchronization primitives `WaitGroup` and `Once`
```
Comment (anzboi) [inline content/gocourse-2019-05-melbourne.slide:847]: Synchronization shouldn't be capitalized
Also feel like there should be a `:` between 'primitives' and 'WaitGroup'

File: content/gocourse-2019-05-melbourne.slide [hunk 2]
```diff
@@ -841,32 +841,16 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
...
+- Intended for low level programming `Mutex`, `RWMutex`, `Cond`
```
Comment (anzboi) [inline content/gocourse-2019-05-melbourne.slide:848]: This reads a little strange as it is
suggestion: Low level synchronization primitives: ...

File: content/gocourse-2019-05-melbourne.slide [hunk 3]
```diff
@@ -841,32 +841,16 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
...
+- Waiting for goroutines to finish [[https://play.golang.org/p/mMOBMJNtVT2][with WaitGroup]]
+- [[https://play.golang.org/p/OUAncJlBbmS][Unmatched `waitgroup.Add()` and `waitgroup.Done()` calls]]
```
Comment (anzboi) [inline content/gocourse-2019-05-melbourne.slide:850]: The bold blocks look odd in this link.

File: content/gocourse-2019-05-melbourne.slide [hunk 4]
```diff
@@ -841,32 +841,16 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
...
+- More `Add()` than `Done()` causes Deadlock
+- More `Done()` than `Add()` causes negative WaitGroup error
+- Synchronization primitives are used for orchestration, Channels are used for communication
```
Comment (anzboi) [inline content/gocourse-2019-05-melbourne.slide:853]: I'm guessing this is meant to lead on into Channels. IMO leave channels information under channels slide.
I also feel like this line would better go as the first line on this slide.

**Review by anzboi — CHANGES_REQUESTED** [review]
I would like to clean up the slide.
The examples are good. Normally I would advise against code that doesn't run, but in this case I think it is fine.

File: samples/goroutines5.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+func say(s string) {
+	defer wg.Done()
+	for i := 0; i < 2; i++ {
+		time.Sleep(100 * time.Millisecond)
+		fmt.Println(s)
+	}
+}
+
+func main() {
+
```
Comment (juliaogris) [inline samples/goroutines5.go:20]: unnecessary blank lines.

**Comment by linuxmonk** [comment]
I am planning to close this PR as we've decided to cover channels.

**Comment by camscale** [comment]
@linuxmonk If this is meant to be closed, could you close it please? It would be nice to get the number of open pull requests down to zero before the course starts today. Thanks.

**Comment by linuxmonk** [comment]
Closing as per earlier discussions

**Pull request closed without merging**